### PR TITLE
benchmarks updates 

### PR DESCRIPTION
**Pull request by PhilipDeegan**
- cleanup of cmake includes/deps/etc

## Summary by CodeRabbit

- **New Features**
  - Enhanced benchmarking configurations and performance testing capabilities.
  - Added new methods for particle data manipulation and simulation benchmarking.
  - Improved support for Python 3.8 and MPI components in project setups.

- **Bug Fixes**
  - Adjusted include paths and namespaces in benchmarking tools for correct functionality.

- **Refactor**
  - Streamlined CMake configurations across modules by removing redundant code.
  - Improved method declarations for clarity and readability in HDF5 file handling.

- **Documentation**
  - Updated internal documentation to reflect configuration and functionality enhancements.

### CODE REVIEW

**Comment by nicolasaunai** [comment]
investigate MPI issues maybe (missing headers?)

**Comment by PhilipDeegan** [comment]
mpicxx link time errors at https://hephaistos.lpp.polytechnique.fr/teamcity/buildConfiguration/Phare_Phare_BuildGithubPr/80127?buildTab=log&linesState=819.2809&logView=flowAware

I don't see this on my host locally so I'll see about reproducing it in the docker image

File: src/core/utilities/point/point.hpp [hunk 1]
```diff
@@ -171,6 +171,7 @@ namespace core
         NO_DISCARD auto end() { return r.end(); }
         NO_DISCARD auto end() const { return r.end(); }
 
+        NO_DISCARD auto& operator*() const { return r; }
```
[the remark this answers is not in the document]
Comment (PhilipDeegan) [reply src/core/utilities/point/point.hpp:174]: In this case it is automatically const-qualified. Stop suggesting to change it

File: src/core/CMakeLists.txt [hunk 2]
```diff
@@ -56,15 +56,8 @@ set( SOURCES_CPP
      utilities/mpi_utils.cpp
     )
 
-find_package(MPI)
-
-add_library(${PROJECT_NAME}  ${SOURCES_INC} ${SOURCES_CPP})
-target_compile_options(${PROJECT_NAME}  PRIVATE ${PHARE_WERROR_FLAGS})
-target_link_libraries(${PROJECT_NAME}  PRIVATE phare_initializer ${MPI_C_LIBRARIES} ${MPI_CXX_LIBRARIES}
```
Comment (PhilipDeegan) [inline src/core/CMakeLists.txt:63]: I have reproduced the mpicxx link error locally and have verified that this removal of `${MPI_CXX_LIBRARIES}` is in fact what is causing the error.

We were always including the mpicxx library before in this lib, or that use this lib, which we don't need anymore

File: tools/bench/core/bench.hpp [hunk 3]
```diff
@@ -116,28 +156,63 @@ auto rho(GridLayout const& layout)
 }
 
 
-
 template<typename GridLayout>
-class Flux : public VecField<GridLayout::dimension>
+class VField : public VecField<GridLayout::dimension>
```
Comment (PhilipDeegan) [inline tools/bench/core/bench.hpp:160]: most vecfields here are redundant with the fixtures added in the views PR

File: tools/bench/core/numerics/interpolator/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,5 @@
+cmake_minimum_required (VERSION 3.9)
```
Comment (nicolasaunai) [inline tools/bench/core/numerics/interpolator/CMakeLists.txt:1]: Some edits in this PR have `(VERSION 3.20.1)`, is that consistent?

File: tools/bench/core/numerics/ion_updater/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,5 @@
+cmake_minimum_required (VERSION 3.9)
```
Comment (nicolasaunai) [inline tools/bench/core/numerics/ion_updater/CMakeLists.txt:1]: same comment about cmake version

**Review by nicolasaunai — APPROVED** [review]

File: tools/bench/core/numerics/interpolator/CMakeLists.txt (hunk 4 above)
Comment (PhilipDeegan) [reply tools/bench/core/numerics/interpolator/CMakeLists.txt:1]: https://github.com/PHAREHUB/PHARE/blob/master/CMakeLists.txt#L1 

cmake was complaining about old versions

**Pull request merged**